datafusion/sql/src/unparser/plan.rs

```diff
@@ -162,10 +162,69 @@ impl Unparser<'_> {
             )]);
         }
 
+        // Construct a list of all the identifiers present in query sources
+        let mut all_idents = Vec::new();
```

Comment:
Might be good to see if this additional logic could be moved to helper function(s)